### PR TITLE
bpo-34797: Convert heapq to the argument clinic

### PR DESCRIPTION
**Pull request by pablogsal**
https://bugs.python.org/issue34797

### CODE REVIEW

**Review by serhiy-storchaka — COMMENTED** [review]
Use more descriptive `object` instead of `'O'`.

File: Modules/_heapqmodule.c
Comment (taleinat) [inline Modules/_heapqmodule.c]: Why was the period at the end of the doc-string removed?

This applies here and in the other max-heap-variant functions.

**Review by taleinat — CHANGES_REQUESTED** [review]
Looking good! Just one minor note.

File: Modules/_heapqmodule.c
Comment (pablogsal) [reply Modules/_heapqmodule.c]: Good catch and thank you @taleinat !

**Comment by pablogsal** [comment]
I have made the requested changes; please review again

**Review by taleinat — APPROVED** [review]

**Comment by taleinat** [comment]
@pablogsal, please wait for @rhettinger's review before merging this in.

**Review by serhiy-storchaka — APPROVED** [review]
LGTM, but these changes need to be approved by @rhettinger.

**Review by auvipy — APPROVED** [review]

**Comment by pablogsal** [comment]
@taleinat @serhiy-storchaka Thanks for the review! I will wait for @rhettinger to approve this change.

**Review by rhettinger — APPROVED** [review]

**Comment by rhettinger** [comment]
Nice work.  Thanks.

**Pull request merged**
